src/libcore/iter/range.rs

```diff
@@ -280,7 +353,8 @@ impl<A: Step> Iterator for ops::RangeFrom<A> {
 
     #[inline]
     fn next(&mut self) -> Option<A> {
-        let mut n = self.start.add_one();
+        // Overflow can happen here. Panic when it does.
+        let mut n = self.start.forward(1).expect("overflow in RangeFrom::next");
```

Comment:
IIRC this used to only panic in debug, right?

I like the change, but it might need to be called out in documentation somewhere.